plugins/woocommerce/includes/class-wc-payment-gateways.php

```diff
@@ -113,6 +118,168 @@ public function init() {
+		 * Allows adding to the addresses that receive payment gateway enabled notifications.
+		 *
+		 * @param array              $email_addresses The array of email addresses to notify.
+		 * @param WC_Payment_Gateway $gateway The gateway that was enabled.
+		 * @return array             The augmented array of email addresses to notify.
+		 * @since 8.5.0
+		 */
+		$email_addresses   = apply_filters( 'wc_payment_gateway_enabled_notification_email_addresses', array(), $gateway );
+		$email_addresses   = array_unique(
+			array_filter(
+				$email_addresses,
+				function( $email_address ) {
+					return filter_var( $email_address, FILTER_VALIDATE_EMAIL );
+				}
+			)
+		);
+		$email_addresses[] = $admin_email;
+
+		$logger = wc_get_container()->get( LegacyProxy::class )->call_function( 'wc_get_logger' );
+		$logger->info( sprintf( 'Payment gateway enabled: "%s"', $gateway_title ) );
+
+		$email_text = sprintf(
+			/* translators: Payment gateway enabled notification email. 1: Username, 2: Gateway Title, 3: Site URL, 4: Gateway Settings URL, 5: Admin Email, 6: Site Name, 7: Site URL. */
+			__(
+				'Howdy %1$s,
+
+The payment gateway "%2$s" was just enabled on this site:
+%3$s
+
+If this was intentional you can safely ignore and delete this email. 
+
+If you did not enable this payment gateway, please log in to your site and consider disabling it here:
+%4$s
+
+This email has been sent to %5$s
+
+Regards,
+All at %6$s
+%7$s',
+				'woocommerce'
+			),
+			$username,
+			$gateway_title,
+			$site_url,
+			$gateway_settings_url,
+			$admin_email,
+			$site_name,
+			$site_url
+		);
+
+		if ( '' !== get_option( 'blogname' ) ) {
+			$site_title = wp_specialchars_decode( get_option( 'blogname' ), ENT_QUOTES );
+		} else {
+			$site_title = wp_parse_url( home_url(), PHP_URL_HOST );
+		}
+
+		return wp_mail(
+			$email_addresses,
+			sprintf(
+				/* translators: Payment gateway enabled notification email subject. %s1: Site title, $s2: Gateway title. */
+				__( '[%1$s] Payment gateway "%2$s" enabled', 'woocommerce' ),
+				$site_title,
+				$gateway_title
+			),
+			$email_text
+		);
+	}
+
+	/**
+	 * Determines from changes in settings if a gateway was enabled.
+	 *
+	 * @param array $value New value.
+	 * @param array $old_value Old value.
+	 * @return bool Whether the gateway was enabled or not.
+	 */
+	private function was_gateway_enabled( $value, $old_value = null ) {
+		if ( null === $old_value ) {
+			// There was no old value, so this is a new option.
+			if ( ! empty( $value ) && is_array( $value ) && isset( $value['enabled'] ) && 'yes' === $value['enabled'] && isset( $value['title'] ) ) {
+				return true;
+			}
+			return false;
+		}
+		// There was an old value, so this is an update.
+		if ( ! empty( $value ) && ! empty( $old_value ) && is_array( $value ) && is_array( $old_value ) && isset( $value['enabled'] ) && isset( $old_value['enabled'] ) && 'yes' === $value['enabled'] && 'yes' !== $old_value['enabled'] && isset( $value['title'] ) ) {
```

Comment:
Optional thing to consider: some of these `if` statements are rather long and hard to read. As ever, it's a matter of trade-offs, but a couple of things we could consider:

- Dropping the `! empty( $value )` checks.
- Using `ArrayUtil`.

Example of the latter:

```php
use Automattic\WooCommerce\Utilities\ArrayUtil;

if ( 
	ArrayUtil::get_value_or_default( (array) $value, 'enabled' ) === 'yes' 
	&& ArrayUtil::get_value_or_default( (array) $old_value, 'enabled' ) !== 'yes' 
) {
	// Huzzah!
}
```

Besides using ArrayUtil, I'm cheating a little by multi-lining things ... but even on one line it's *arguably* easier to parse and get the meaning at a glance (subjective, I know).

Reply:
Great call -- that looks much better. 💯